code-challenges.test.js

```diff
@@ -36,17 +44,69 @@ const object3 = { number: -7 }
+        //output: number is/is not, divisible by 3
+            
+
+const divisibleByThree = (obj) => {
+    let number = obj.number
+    if (number % 3 === 0) {
+        return `${number} is divisible by three`
+    } else {
+        return `${number} is not divisible by three`
+    }
+}  
+console.log(divisibleByThree(object1));
 // Explain your code:
+    //takes an object as a parameter
+            //created a variable named number that takes the numerical value from the object
+            //the if statement compairs the values and checks to see if the numbers are divisible by 3 and returns the appropriate statement
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
+describe("allFirstCaps", () => {
+    it("returns an array with all the words capitalized at the 0 indes", () => {
+        expect(allFirstCaps(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(allFirstCaps(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+// ReferenceError: allFirstCaps is not defined
 
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
-
-// Explain your code:
+    //Psudo code
+        //create function that takes in array as parameter
+        //create variable to store new array
+        //for loop to iterate over each word in array
+        //use charAt to start at 0 index
+        //use toUpperCase to capatalize that index
+        //use slice to create new string
+        //use toLowerCase to make the rest of the words lower case
+        //output "Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+            // ["Temperature", "Database", "Chopsticks", "Mango"]
+const allFirstCaps = (arr) => {
+    let capWords = [];
+    for (let i = 0; i < arr.length; i++) {
+    let word = arr[i];
+    let capWord = word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
+        capWords.push(capWord)
+    }
+    return capWords
+}        
+// console.log(allFirstCaps(randomNouns2));
+// Explain your code:   
+    //create function that takes in parameter
+    //created variable capWords to hold new array
+    //used a for loop to iterate over each word in the array
+    //created word variable to get the current word at that index
+    //used charAt, toUpperCase, slice and toLowercase to capatalize the first letter at the 0 index and convert the rest of the word to lowercase
+    //used the .push method to add the capitalized word to the array
```

Comment:
Good work!  I encourage you to try to refactor using HoFs.  You can totally do it.